sympy/discrete/convolution.py

```diff
@@ -115,7 +119,7 @@ def convolution_fft(a, b, dps=None):
 
     a = ifft(a, dps)[:m]
 
-    return a
+    return a if c is None else [sum(a[i::c]) for i in range(c)]
```

Comment:
This operation is essentially independent of implementation. Could this be handled by the main function `convolution`? It would not be necessary to pass `cycle` to implementation functions like `convolution_fft` and `convolution_ntt`.

Reply:
Yes, that could be done, but the specialized functions are also public - wouldn't they be requiring an extra argument for cyclic convolution?

Reply:
They can be public but I think that they can be specialized to a single operation. `convolution` should be the function that most users would need.

Reply:
Okay, so cyclic should be available only to `convolution`?

Reply:
I think that that would suffice, at least for now.